src/main/Core/ImageGenerator/LinuxIconExtractor.ts

```diff
@@ -0,0 +1,354 @@
+import type { CommandlineUtility } from "@Core/CommandlineUtility";
+import type { FileSystemUtility } from "@Core/FileSystemUtility";
+import type { IniFileParser } from "@Core/IniFileParser/IniFileParser";
+import type { Logger } from "@Core/Logger";
+import type { Image } from "@common/Core/Image";
+import { dirname, extname, join } from "path";
+import type { CacheFileNameGenerator } from "./CacheFileNameGenerator";
+import type { FileIconExtractor } from "./FileIconExtractor";
+
+// Specs from https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
+interface IconThemeSubdir {
```

Comment:
Please use types instead of Interfaces:

```suggestion
type IconThemeSubdir = {
```